### DISCUSSION TITLE
Get rectangle corners from a Plane3dByOriginAndUnitNormal and a center

### DISCUSSION THREAD

**Discussion by salah-walid** [iTwin/itwinjs-core#4403] [Q&A] [answered]
Hello

Having a Plane3dByOriginAndUnitNormal and a center point in that plane also rectangle width and height, is there a way to get rectangle points in that plane.

Thank you in advance

**Comment by bbastings** [comment]
The rotation of the rectangle in the plane is not clearly defined. You just have an up vector, so there are an infinite number of perpendicular vectors.

You can use a method like Matrix3d.createRigidHeadsUp(plane.getNormalRef()) to get some x and y axis to use to offset the center point in order to get the corners of "a" rectangle in the plane...

**Reply by salah-walid** [reply]
Having that matrix3D how can somone translate that point knowing the distance and the translation vector ?

**Comment by bbastings** [comment, marked as answer]
One approach would be something like this:

// Create unrotated rectangle centered at 0,0,0 in XY plane...
const shapePts: Point3d[] = [];
shapePts[0] = Point3d.create(-halfWidth, -halfHeight);
shapePts[1] = Point3d.create(halfWidth, -halfHeight);
shapePts[2] = Point3d.create(halfWidth, halfHeight);
shapePts[3] = Point3d.create(-halfWidth, halfHeight);
shapePts[4] = shapePts[0].clone();

// Transform shape points...
const transform = Transform.createOriginAndMatrix(center, matrix);
transform.multiplyPoint3dArrayInPlace(shapePts);
